### PR TITLE
Implement spend test to label in http integration test

### PR DESCRIPTION
**Pull request by sennui**

### CODE REVIEW

**Comment by lucafavatella** [comment]
In CI push (not PR) build I see [failure](https://travis-ci.org/aeternity/epoch/jobs/336474588#L1431) unrelated to this PR:
```
%%% aec_eunit_SUITE ==> {error,{errors_in_lager_log,["09:36:56.600 [error] Supervisor exec_app had child exec started with exec:start_link([]) at <0.20899.0> exit with reason killed in context shutdown_error\n\n"]}}
```

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 1]
```diff
@@ -204,9 +205,9 @@ groups() ->
 
         naming_system_manage_name
       ]},
+     {naming, [sequence], [naming_system_manage_name]},
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: I see this test also run as internal_endpoints. No issues with that.

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 2]
```diff
@@ -1470,10 +1474,23 @@ naming_system_manage_name(_Config) ->
...
-    TmpPointers = <<"{\"account_pubkey\":\"srtrst\"}">>,
     {ok, 200, #{<<"name">>     := Name,
                 <<"name_ttl">> := NameTTL,
-                <<"pointers">> := TmpPointers}} = get_name(Name),
+                <<"pointers">> := Pointers}} = get_name(Name),
+
+    {ok, 200, #{<<"balance">> := Balance3}} = get_balance(),
+    Host = internal_address(),
+    http_request(Host, post, "spend-tx",
+        #{recipient_pubkey => Name,
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: I see this is the main point of the test.

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 3]
```diff
@@ -1470,10 +1474,23 @@ naming_system_manage_name(_Config) ->
...
     %% Check that TTL and pointers got updated in name entry
-    TmpPointers = <<"{\"account_pubkey\":\"srtrst\"}">>,
     {ok, 200, #{<<"name">>     := Name,
                 <<"name_ttl">> := NameTTL,
-                <<"pointers">> := TmpPointers}} = get_name(Name),
+                <<"pointers">> := Pointers}} = get_name(Name),
+
+    {ok, 200, #{<<"balance">> := Balance3}} = get_balance(),
+    Host = internal_address(),
+    http_request(Host, post, "spend-tx",
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Checking return code would be nice here. Though optional for this PR.

File: apps/aehttp/test/aehttp_integration_SUITE.erl (hunk 2 above)
Comment (sennui) [reply apps/aehttp/test/aehttp_integration_SUITE.erl]: yes, correct

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 4]
```diff
@@ -1470,10 +1474,23 @@ naming_system_manage_name(_Config) ->
...
+        #{recipient_pubkey => Name,
+          amount => 77,
+          fee => 50}),
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE), 1),
+
+    %% Nothing gets lost as recipient = sender = miner
+    %% This tests 'resolve_name' because recipient is expressed by name label
+    %% This tests passes with 1 block confirmation due to lack of miner's reward delay
+    FinalBalance = Balance3+MineReward,
+    {ok, 200, #{<<"balance">> := FinalBalance}} = get_balance(),
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Test does not guarantee that spend tx included in block.
* Return code of spend tx HTTP call not checked;
* Mempool not checked after post spend tx;
* Balance post-mining is the same regardless spend tx was included or not.

**Pull request merged**